Content.Shared/_Starlight/Antags/Abductor/EntitySystems/AbductorWonderprodSystem.cs

```diff
@@ -0,0 +1,82 @@
+using Content.Shared._Starlight.Antags.Abductor.Components;
+using Content.Shared._Starlight.Clothing.Components;
+using Content.Shared.Damage.Components;
+using Content.Shared.Inventory;
+using Content.Shared.Weapons.Melee.Events;
+using Robust.Shared.Timing;
+
+namespace Content.Shared._Starlight.Antags.Abductor.EntitySystems;
+
+/// <summary>
+/// System that handles abductor wonderprod interactions with hardsuit immunity.
+/// When a wonderprod hits someone with hardsuit protection, it reduces its stamina damage to stunbaton levels instead of full wonderprod damage.
+/// </summary>
+public sealed class AbductorWonderprodSystem : EntitySystem
+{
+    [Dependency] private readonly InventorySystem _inventory = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        
+        // Subscribe to melee hit events from wonderprods to check for hardsuit immunity
+        SubscribeLocalEvent<AbductorWonderprodComponent, MeleeHitEvent>(OnWonderprodMeleeHit);
+    }
+
+    private void OnWonderprodMeleeHit(Entity<AbductorWonderprodComponent> ent, ref MeleeHitEvent args)
+    {
+        // Check if the wonderprod has a StaminaDamageOnHit component
+        if (!TryComp<StaminaDamageOnHitComponent>(ent, out var staminaDamage))
+            return;
+
+        // Store the original damage value
+        var originalDamage = staminaDamage.Damage;
+        var hasHardsuitTargets = false;
+
+        // Check each target for hardsuit immunity
+        foreach (var target in args.HitEntities)
+        {
+            if (HasHardsuitImmunity(target))
+            {
+                hasHardsuitTargets = true;
```

Comment:
question: does this make it so if someone wearing a hardsuit is hit it reduces stam dammage to all hit targets or ???